src/v/redpanda/application.cc

```diff
@@ -314,23 +316,69 @@ void application::initialize(
 }
 
 void application::setup_metrics() {
-    if (!config::shard_local_cfg().disable_public_metrics()) {
-        seastar::metrics::replicate_metric_families(
-          seastar::metrics::default_handle(),
-          {{"scheduler_runtime_ms", ssx::metrics::public_metrics_handle},
-           {"io_queue_total_read_ops", ssx::metrics::public_metrics_handle},
-           {"io_queue_total_write_ops", ssx::metrics::public_metrics_handle},
-           {"memory_allocated_memory", ssx::metrics::public_metrics_handle},
-           {"memory_free_memory", ssx::metrics::public_metrics_handle}})
-          .get();
-    }
+    setup_internal_metrics();
+    setup_public_metrics();
+}
 
-    if (config::shard_local_cfg().disable_metrics()) {
+void application::setup_public_metrics() {
+    namespace sm = ss::metrics;
+
+    if (config::shard_local_cfg().disable_public_metrics()) {
         return;
     }
 
+    seastar::metrics::replicate_metric_families(
+      seastar::metrics::default_handle(),
+      {{"io_queue_total_read_ops", ssx::metrics::public_metrics_handle},
+       {"io_queue_total_write_ops", ssx::metrics::public_metrics_handle},
+       {"memory_allocated_memory", ssx::metrics::public_metrics_handle},
+       {"memory_free_memory", ssx::metrics::public_metrics_handle}})
+      .get();
+
+    _public_metrics.start().get();
+
+    _public_metrics
+      .invoke_on(
+        ss::this_shard_id(),
+        [](auto& public_metrics) {
```

Comment:
@VladLazar just curious about using `invoke_on(this_shard_id, ...` as opposed to `_public_metrics.local(). ..` is there something special about using invoke on that i'm missing here?

Reply:
You're not missing anything. I just slightly misused the api. Should have used `local()`. I don't think there is any difference in behaviour though.

Reply:
me neither, looks fine. thx